projects/canopy/src/lib/forms/sort-code/sort-code.component.ts

```diff
@@ -158,18 +158,32 @@ export class LgSortCodeComponent implements OnInit, ControlValueAccessor {
   }
 
   validate(): ValidationErrors {
-    const errors: ValidationErrors = {};
+    const invalidFields: Array<string> = [];
     Object.keys(this.sortCodeFormGroup.controls).forEach((fieldName) => {
       if (
         this.sortCodeFormGroup.controls[fieldName].invalid &&
         (this.sortCodeFormGroup.controls[fieldName].dirty ||
           (this.formGroupDirective && this.formGroupDirective.submitted))
       ) {
-        errors.invalidField = true;
+        invalidFields.push(fieldName);
       }
     });
-    if (Object.keys(errors).length) {
-      return errors;
+
+    const requiredFields = invalidFields.filter((fieldName) =>
+      this.sortCodeFormGroup.controls[fieldName].hasError('required'),
+    );
+
+    const error: ValidationErrors = {};
+    // If all the fields have been left empty we show the
```

Comment:
I'd be tempted to lose the comment as I feel the code does a good job of explaining it.

Reply:
Happy to remove that

Reply:
Removed